pepper-apis/src/main/java/org/broadinstitute/ddp/model/event/UpdateCustomWorkflowEventAction.java

```diff
@@ -0,0 +1,54 @@
+package org.broadinstitute.ddp.model.event;
+
+import static org.broadinstitute.ddp.util.GsonCreateUtil.createJsonIgnoreNulls;
+
+import org.broadinstitute.ddp.db.dto.EventConfigurationDto;
+import org.broadinstitute.ddp.event.publish.EventActionPublisher;
+import org.broadinstitute.ddp.model.activity.types.EventActionType;
+import org.broadinstitute.ddp.pex.PexInterpreter;
+import org.jdbi.v3.core.Handle;
+
+/**
+ * Event {@link EventActionType#UPDATE_CUSTOM_WORKFLOW} handler.
+ * It publishes the event to other consumers (with using an instance of
+ * {@link EventActionPublisher} passed in the constructor.
+ * A default implementation of a publisher sends a message to a specified pubsub-topic
+ * (to be consumed by DSM).
+ */
+public class UpdateCustomWorkflowEventAction extends EventAction {
+
+    public static final String PAYLOAD_FIELD__WORKFLOW = "workflow";
+    public static final String PAYLOAD_FIELD__STATUS = "status";
+
+    private static final EventActionType EVENT = EventActionType.UPDATE_CUSTOM_WORKFLOW;
+
+    private final EventActionPublisher eventActionPublisher;
+    private final String workflow;
+    private final String status;
+
+    public UpdateCustomWorkflowEventAction(
+            EventConfiguration eventConfiguration,
+            EventConfigurationDto dto,
+            EventActionPublisher eventActionPublisher) {
+        super(eventConfiguration, dto);
+        this.workflow = dto.getCustomWorkflowName();
+        this.status = dto.getCustomWorkflowStatus();
+        this.eventActionPublisher = eventActionPublisher;
+    }
+
+    @Override
+    public void doAction(PexInterpreter pexInterpreter, Handle handle, EventSignal signal) {
+        eventActionPublisher.publishEventAction(
+                EVENT,
```

Comment:
I don't like how the "pubsub task" we're pulishing is an "event action" type. We shouldn't EventActionType into this. What we should be publishing is a "pubsub task" type. For example, we currently have `UPDATE_PROFILE` type. We should have another type such as `UPDATE_CUSTOM_WORKFLOW`. Although they have the same name, we shouldn't reuse EventActionType like this.... imagine someone passing in an EventActionType that DSM don't even support!

Also, any chance we can refactor the "pubsub task" thing? I had a hard time digging through the code to figure out what "pubsub task" types we support... Will be great if we had an enum somewhere.

Reply:
I don't see a possibility to reuse "pubsub task" because it is used for opposite purpose: to consume tasks created by DSM for DSS (actually it can be not only tasks from DSM but from any other external application which can put tasks to the "pubSubTaskTopic", but anyway this PubSubTask service intended to consume and process tasks assigned for DSS by outside applications).
In this current ticket the similar task consuming service should be implemented on DSM side and on DSS side I implement a client side publishing task for DSM.
I suggest to design this client a more generic way: replace `EventActionType` with a more generic parameter: String dsmTask.